routers/web/repo/compare.go

```diff
@@ -228,10 +228,31 @@ func ParseCompareInfo(ctx *context.Context) *common.CompareInfo {
 	)
 
 	infoPath = ctx.PathParam("*")
+
 	var infos []string
+
 	if infoPath == "" {
 		infos = []string{baseRepo.DefaultBranch, baseRepo.DefaultBranch}
 	} else {
+		// check if head is a branch or tag only if infoPath ends with .diff or .patch
+		if strings.HasSuffix(infoPath, ".diff") || strings.HasSuffix(infoPath, ".patch") {
+			infos = strings.SplitN(infoPath, "...", 2)
+			if len(infos) != 2 {
+				infos = strings.SplitN(infoPath, "..", 2) // match github behavior
+			}
+			ref2IsBranch := gitrepo.IsBranchExist(ctx, ctx.Repo.Repository, infos[1])
+			ref2IsTag := gitrepo.IsTagExist(ctx, ctx.Repo.Repository, infos[1])
```

Comment:
The path format is like this:

What if the path is `{:baseBranch}...{:headOwner}/{:headRepoName}:{:headBranch}.patch`?

![image](https://github.com/user-attachments/assets/2cb4fbee-6bce-45be-912a-56e7bb4bf6a9)